src/component/JoinList.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { withFirestore } from "react-firestore";
+import { Redirect } from 'react-router-dom'
```

Comment:
Nice use of react tools to redirect! Ethan and I ended up using vanilla JS to achieve this, aka window.location.reload(false), but I like this way more.